src/uu/mv/src/mv.rs

```diff
@@ -726,3 +740,122 @@ fn is_empty_dir(path: &Path) -> bool {
         Err(_e) => false,
     }
 }
+
+/// Moves a directory from one location to another with progress tracking.
+///
+/// This is a modified version of `fs_extra::dir::move_dir_with_progress` tailored to fit our use case.
+/// This function assumes that `from` is a directory and `to` does not exist.
+///
+/// Parameters:
+/// - `from`: The source directory path.
+/// - `to`: The destination directory path.
+/// - `progress_bar`: An optional reference to a `ProgressBar` for tracking the movement progress.
+/// - `moved_files`: A mutable HashMap containing information about moved files.
+///
+/// Returns:
+/// - Result<u64>: The total number of bytes moved if successful.
+pub fn move_dir(
```

Comment:
why  pub ? :)
please add unit tests for this function 
and
move_file

Reply:
Ohh! i will do it ASAP